src/common/BrowserStorage.tsx

```diff
@@ -248,26 +319,85 @@ class _BrowserStorage {
 			},
 		};
 		const values = await BrowserStorage.get('options');
-		for (const option of Object.values(options)) {
+		if (values.options) {
+			this.options = values.options;
+		}
+		for (const option of Object.values(this.optionsDetails)) {
 			option.name = I18N.translate(`${option.id}Name` as MessageName);
 			if (!option.description) {
 				option.description = I18N.translate(`${option.id}Description` as MessageName);
 			}
-			option.value = (values.options && values.options[option.id]) || option.value;
+			option.value =
+				typeof this.options[option.id] !== 'undefined' ? this.options[option.id] : option.value;
 			if (option.id === 'streamingServices') {
 				const missingServices = Object.fromEntries(
-					Object.keys(streamingServices)
-						.filter((serviceId) => !(serviceId in option.value))
-						.map((serviceId) => [serviceId, false])
-				) as Record<StreamingServiceId, boolean>;
+					(Object.keys(streamingServices) as StreamingServiceId[])
+						.filter(
+							(serviceId) =>
+								!(serviceId in option.value) || typeof option.value[serviceId] === 'boolean'
+						)
+						.map((serviceId) => [
+							serviceId,
+							{
+								scrobble: false,
+								sync: false,
+								autoSync: false,
+								autoSyncDays: 7,
+								lastSync: 0,
+								lastSyncId: '',
+							},
+						])
+				) as Record<StreamingServiceId, StreamingServiceValue>;
 				option.value = { ...option.value, ...missingServices };
 			}
+			this.addOption(option);
+		}
+	};
+
+	saveOptions = async (options: Partial<StorageValuesOptions>) => {
+		for (const [id, value] of Object.entries(options) as [
+			keyof StorageValuesOptions,
+			StorageValuesOptions[keyof StorageValuesOptions]
+		][]) {
+			this.addOption({ id, value });
+		}
+		await BrowserStorage.set({ options: this.options }, true);
+	};
+
+	addOption = <K extends keyof StorageValuesOptions>(option: Partial<Option<K>>) => {
+		if (typeof option.id !== 'undefined' && typeof option.value !== 'undefined') {
+			if (BrowserStorage.isStreamingServiceOption(option)) {
+				for (const [id, value] of Object.entries(option.value) as [
+					StreamingServiceId,
+					StreamingServiceValue
+				][]) {
+					if (!this.options[option.id]) {
+						this.options[option.id] = {} as Record<StreamingServiceId, StreamingServiceValue>;
+					}
+					this.options[option.id][id] = {
+						...(this.options[option.id]?.[id] ?? {}),
+						...value,
+					};
+					this.optionsDetails[option.id].value[id] = {
+						...this.optionsDetails[option.id].value[id],
+						...value,
+					};
+				}
+			} else {
+				this.options[option.id] = option.value;
+				this.optionsDetails[option.id].value = option.value;
```

Comment:
I've pinpointed the previously mentioned error to this line, when option.id = "disableScrobbling". The optionDetails-lookup object is then undefined. The extension seems to remember enabling/disabling though.

Reply:
Ah, I see, it's because `disableScrobbling` was removed. I just pushed a commit that upgrades/downgrades the storage if necessary, to avoid conflicts like this.